### PR TITLE
Added @tonic-ui/react__icons and updated @tonic-ui/react types for v2

### PR DESCRIPTION
**Pull request by ZachLegros**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/trendmicro-frontend/tonic-ui/releases/tag/%40tonic-ui%2Freact%402.0.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/tonic-ui__react-icons/scripts/generate-definitions/tsconfig.json
1. types/tonic-ui__react-icons/tonic-ui__react-icons-tests.tsx
1. types/tonic-ui__react-icons/index.d.ts
1. types/tonic-ui__react-icons/scripts/generate-definitions/index.ts
1. types/tonic-ui__react/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 67af91181c42756e0c5f381fe3a63ae40167b88e
</p>

**Review by derekhawker — APPROVED** [review]

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/tonic-ui__react-icons/scripts/generate-definitions/package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "private": true,
+    "description": "Script to generate the type definitions for icons in the @trendmicro/tmicon package",
+    "author": "Zachary Legros",
+    "devDependencies": {
+        "@types/node": "*",
+        "@trendmicro/tmicon": "1.40.0"
```
[the remark this answers is not in the document]
Comment (ZachLegros) [reply types/tonic-ui__react-icons/scripts/generate-definitions/package.json:7]: This is a script `package.json` and it looks like other `@types` packages are allowed to add dev dependencies. For example, here is the `@types/node` package using external packages: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0b234975014c1da99b60104b8e5febb7fbcada38/types/node/scripts/generate-docs/package.json#L11

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/tonic-ui__react/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f6db9a93d51fa6f457e302edc6120976285533f8
</p>

File: types/tonic-ui__react-icons/scripts/generate-definitions/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
[the remark this answers is not in the document]
Comment (sheetalkamat) [reply types/tonic-ui__react-icons/scripts/generate-definitions/.eslintrc.json:1]: Please remove and fix errors

File: types/tonic-ui__react-icons/scripts/generate-definitions/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakebailey) [reply types/tonic-ui__react-icons/scripts/generate-definitions/package.json:7]: For now, this is fine.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Review by sheetalkamat — APPROVED** [review]

**Comment by ZachLegros** [comment]
Ready to merge

**Pull request merged**